numba/core/typed_passes.py

```diff
@@ -675,10 +675,16 @@ def _run_inliner(
             arg_typs = template._inline_overloads[arg_typs]['folded_args']
             iinfo = template._inline_overloads[arg_typs]['iinfo']
             freevars = iinfo.func_ir.func_id.func.__code__.co_freevars
+            # operate on the closest thing we have to a copy
+            # the IR is cached on the overload template
+            # the inliner working parts are notorious for mutating IR directly
+            blocks_copy = {k: v.copy() for k, v in iinfo.func_ir.blocks.items()}
+            func_ir_copy = iinfo.func_ir.copy()
+            func_ir_copy.blocks = blocks_copy
```

Comment:
I don't think the `blocks` need to be copied explicitly. `func_ir.copy()` has already done that. 

```suggestion
            func_ir_copy = iinfo.func_ir.copy()
```

the test passes with the change.

Reply:
Good point, will patch.

Reply:
I ended up having to do a much more involved copy.